app/src/scrcpy.c

```diff
@@ -102,6 +104,18 @@ static void event_loop(void) {
             case SDL_MOUSEBUTTONUP:
                 input_manager_process_mouse_button(&input_manager, &event.button);
                 break;
+            case SDL_DROPFILE:
+
+                if (!installer.initialized) {
+                    SDL_bool init_ok = installer_init(&installer, server.serial);
+                    if (init_ok && installer_start(&installer)) {
+                        goto push;
+                    }
+                    installer_destroy(&installer);
+                }
```

Comment:
There is a logic flaw here: if initialization fails, it still calls `installer_push_apk()`.